src/goto-symex/symex_builtin_functions.cpp

```diff
@@ -184,6 +184,12 @@ void goto_symext::symex_allocate(
     else
       throw "failed to zero initialize dynamic object";
   }
+  else
+  {
+    exprt nondet = build_symex_nondet(object_type);
```

Comment:
Point of interest: you've copied the case for `x = NONDET(int)` rather than the case from `code_declt("x", java_int_type())` for example. Check how their behaviour differs if at all.

Reply:
It'll only disable it for the time being, but it's interesting. I'm not quite well informed enough to give an authoritative answer -- I suggest running this by @peterschrammel and post his answer here.

Reply:
I'd argue that this is what #35 is about, and you're doing parts of that here, even though the full value only unfolds once you break down nondet symbols into their individual fields.